app/core/contents.js

```diff
@@ -32,105 +28,115 @@ function handler (activePageSlug, config) {
     files: []
   });
 
-  files.forEach(filePath => {
+  const results = await Promise.all(
+    files.map(filePath => processFile(config, activePageSlug, contentDir, filePath))
```

Comment:
Most of the code went into the `processFile` async function that gets fired in parallel. Sadly it's not easy to diff this change but it was necessary for clarity.
You'll see that after this finishes the results get feed into `filesProcessed` and afterwards sorted just like before.